mobile-app/src/firebase/api.ts

```diff
@@ -96,73 +110,102 @@ export async function fetchHabitCompletionsForParticipant({
 }
 
 // FRIENDS
-export async function fetchFriends(): Promise<allUsersInfoT> {
-  const userId = atomStore.get(currentUserIdAtom);
-
+export async function fetchFriendIds({
+  userId,
+}: {
+  userId: string;
+}): Promise<string[]> {
+  const friendshipsCollection = collection(firestore, "friendships");
+  const q = query(
+    friendshipsCollection,
+    or(where("user1Id", "==", userId), where("user2Id", "==", userId)),
+  );
+  const myFriendshipSnap = await getDocs(q);
   const myFriendIds: string[] = [];
-  for (const friendshipId in mockFriendships) {
-    const friendship = mockFriendships[friendshipId];
+  myFriendshipSnap.forEach((doc) => {
+    const friendship = doc.data() as friendshipT;
     if (friendship.user1Id === userId) {
       myFriendIds.push(friendship.user2Id);
     } else if (friendship.user2Id === userId) {
       myFriendIds.push(friendship.user1Id);
     }
-  }
+  });
+  return myFriendIds;
+}
 
-  return Object.fromEntries(
-    Object.entries(mockUsers).filter(([userId]) =>
-      myFriendIds.includes(userId),
-    ),
-  );
+export async function fetchFriendData({
+  userId,
+}: {
+  userId: string;
+}): Promise<allUsersInfoT> {
+  const allFreindData: allUsersInfoT = {};
+  const myFriendIds = await fetchFriendIds({ userId });
+  for (const friendId of myFriendIds) {
+    const userDocRef = doc(firestore, "users", friendId);
+    const userDocSnap = await getDoc(userDocRef);
+    if (userDocSnap.exists()) {
+      const userData = userDocSnap.data() as userT;
+      allFreindData[friendId] = {
+        createdAt: userData.createdAt,
+        displayName: userData.displayName,
+        username: userData.username,
+        picture: userData.picture,
+      };
+    } else {
+      console.error(`No user found with ID: ${friendId}`);
+    }
+  }
+  return allFreindData;
 }
 
-export async function fetchCommonHabits({
+export async function fetchCommonHabitIds({
   participantId,
 }: {
   participantId: string;
 }): Promise<string[]> {
   const userId = atomStore.get(currentUserIdAtom);
-  // fetch habits WHERE userId in participants AND participantId in participants
-  return Object.keys(mockHabits).filter(
-    (habitId) =>
-      mockHabits[habitId].participants[userId] &&
-      mockHabits[habitId].participants[participantId],
+  const habitsCollection = collection(firestore, "habits");
+  const userQuery = query(
+    habitsCollection,
+    where(`participants.${userId}`, "!=", null),
   );
+  const userHabitsSnap = await getDocs(userQuery);
+  const commonHabitIds: string[] = [];
+  userHabitsSnap.forEach((doc) => {
+    const habitData = doc.data();
+    if (habitData.participants[participantId]) {
+      commonHabitIds.push(doc.id);
+    }
+  });
+  return commonHabitIds;
 }
 
 export async function fetchMutualFriends({
   friendId,
 }: {
   friendId: string;
 }): Promise<allUsersInfoT> {
```

Comment:
To fix the cyclic dependency, you can make it so that this function also takes in friendIds. That way we don't need to use that Jotai atom here. Then you can also probably get rid of friendsAtom.ts and keep everything in atoms.ts